store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/handler/spark/AddElementsFromRDD.java

```diff
@@ -53,52 +55,94 @@
 public class AddElementsFromRDD {
     private static final Logger LOGGER = LoggerFactory.getLogger(AddElementsFromRDD.class);
 
-    void addElementsFromRDD(final JavaRDD<Element> input, final Context context, final ParquetStore store) throws OperationException {
-        final FileSystem fs = store.getFS();
-        final ParquetStoreProperties storeProperties = store.getProperties();
-        final SchemaUtils schemaUtils = store.getSchemaUtils();
-        final SparkSession spark = SparkContextUtil.getSparkSession(context, store.getProperties());
-        SparkParquetUtils.configureSparkForAddElements(spark, storeProperties);
-        final String tempDir = store.getProperties().getTempFilesDir();
+    private final ParquetStore store;
+    private final Schema schema;
+    private final SchemaUtils schemaUtils;
+    private final FileSystem fs;
+    private final SparkSession spark;
+    private final String tempDir;
+    private final Function<String, String> groupToUnsortedUnaggregatedNewData;
```

Comment:
These functions should probably be replaced with a single method call which take group, sorted and reversed arguments.